core/time/shared/tz.rb

```diff
@@ -0,0 +1,39 @@
+describe :time_tz, shared: true do
+  describe "TZ env variable" do
+    before do
+      @tz = ENV["TZ"]
+
+      @get_time = @object
+    end
+
+    after do
+      ENV["TZ"] = @tz
+    end
+
+    it "returns time in timezone specified by TZ env variable" do
+      ENV["TZ"] = "Asia/Hong_Kong"
+      @get_time.call.zone.should == "HKT"
+
+      ENV["TZ"] = "Africa/Dakar"
+      @get_time.call.zone.should == "GMT"
+    end
+
+    it "returns time in UTC if specified incorrect value" do
+      #ENV["TZ"] = "bla-bla-bla"
+      #@get_time.call.zone.should == "UTC"
+
+      ENV["TZ"] = ""
+      @get_time.call.zone.should == "UTC"
+
+      ENV["TZ"] = "  "
+      @get_time.call.zone.should == "UTC"
+    end
+
+    it "returns time in local timezone if specified nil" do
+      local_zone = @get_time.call.zone
+
+      ENV["TZ"] = nil
+      @get_time.call.zone.should == local_zone
```

Comment:
This spec would fail if `TZ` is set to a zone different than the local one while running the specs.
I think we cannot test much here, except that it should return a zone, even if `TZ` is unset.

Reply:
@eregon Yeah, you are right